Minio/DataModel/ObjectOperationsArgs.cs

```diff
@@ -0,0 +1,142 @@
+/*
+ * MinIO .NET Library for Amazon S3 Compatible Cloud Storage, (C) 2020 MinIO, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+using RestSharp;
+using System;
+using System.Collections.Generic;
+using System.Xml.Linq;
+
+namespace Minio
+{
+    public class RemoveObjectArgs : ObjectArgs<RemoveObjectArgs>
+    {
+        public string VersionId { get; private set; }
+
+        public RemoveObjectArgs()
+        {
+            this.RequestMethod = Method.DELETE;
+        }
+
+        public override RestRequest BuildRequest(RestRequest request)
+        {
+            if (!string.IsNullOrEmpty(this.VersionId))
+            {
+                request.AddQueryParameter("versionId",$"{this.VersionId}");
+            }
+            return request;
+        }
+        public RemoveObjectArgs WithVersionId(string ver)
+        {
+            this.VersionId = ver;
+            return this;
+        }
+    }
+
+    public class RemoveObjectsArgs : ObjectArgs<RemoveObjectsArgs>
+    {
+        internal List<string> ObjectNames { get; private set; }
+        // Each element in the list is a Tuple. Each Tuple has an Object name(string)
+        // And the second element is 
+        internal List<Tuple<string, List<string>>> ObjectNamesVersions  { get; private set; }
+
+        public RemoveObjectsArgs()
+        {
+            this.ObjectName = null;
+            this.ObjectNames = new List<string>();
+            this.ObjectNamesVersions = new List<Tuple<string, List<string>>>();
```

Comment:
For  List<Tuple<string,string>>. My thinking was if there is one single object & multiple versions, then the user will have to enter the object name repeatedly. So, I used  List<Tuple<string, List<string>>>() instead. Tuple<object_name, List<multiple_version_ids>>. I can change it to List<Tuple<string,string>>.

Reply:
you can still provide a .WithVersionIDs() method for the RemoveObjectAsync api call - and internally convert the args to a List<Tuple<string,string>> so that user doesn't have to repeat the objectName.  Otherwise the implementation looks needlessly complicated to batch the calls if a List<Tuple<string, List>> is used

Reply:
Changed to this Tuple.